### PR TITLE
Fix borrow balance and borrow limit calculations

### PR DESCRIPTION
**Pull request by johnmosesman**

### CODE REVIEW

File: app/components/borrow-flow/borrow.tsx [hunk 1]
```diff
@@ -49,21 +50,22 @@ export default function Borrow({
...
+    totalBorrowedAmountInUsd,
+    +value
   );
 
   let newBorrowLimitUsed = useBorrowLimitUsed(
-    totalBorrowedAmountInUsd,
-    newBorrowLimit
+    newTotalBorrowedAmountInUsd,
+    borrowLimit
   );
```
Comment (johnmosesman) [inline app/components/borrow-flow/borrow.tsx:69]: When borrowing the borrow limit doesn't change, that only happens when supplying/withdrawing assets. The part that changes is how much you're borrowing against that limit.

File: app/components/borrow-flow/borrow.tsx [hunk 2]
```diff
@@ -49,21 +50,22 @@ export default function Borrow({
...
+  let newTotalBorrowedAmountInUsd = useNewTotalBorrowedAmountInUsd(
     signer,
-    market.comptrollerAddress,
-    tokenPairs,
-    market.tokenPair.cToken,
-    `-${value}`
+    market.tokenPair,
+    totalBorrowedAmountInUsd,
+    +value
   );
```
Comment (johnmosesman) [inline app/components/borrow-flow/borrow.tsx:64]: Borrow limit doesn't change. This new hook calculates new borrow amount + USD conversion which was missing before.

File: app/components/borrow-flow/repay.tsx [hunk 3]
```diff
@@ -53,17 +55,16 @@ export default function Repay({
...
+    totalBorrowedAmountInUsd,
+    -value
   );
 
   let newBorrowLimitUsed = useBorrowLimitUsed(
-    totalBorrowedAmountInUsd,
-    newBorrowLimit
+    newTotalBorrowedAmountInUsd,
+    borrowLimit
   );
```
Comment (johnmosesman) [inline app/components/borrow-flow/repay.tsx:68]: Same comments here as in borrow.tsx

File: app/lib/tender.ts [hunk 4]
```diff
@@ -288,16 +288,24 @@ async function projectBorrowLimit(
...
+    tokenPair
   );
 
-  // Collateral factors are always 1e18
-  let collateralFactor: number = parseFloat(
-    formatUnits(rawCollateralFactor, 18)
+  let priceInUsd = await getAssetPriceInUsd(
+    signer,
+    tokenPair.token.priceOracleAddress
   );
```
Comment (johnmosesman) [inline app/lib/tender.ts:300]: Price in USD was missing from the calculation

File: app/components/borrow-flow/repay.tsx [hunk 5]
```diff
@@ -53,17 +55,16 @@ export default function Repay({
...
-  let newBorrowLimit = useProjectBorrowLimit(
+  let newTotalBorrowedAmountInUsd = useNewTotalBorrowedAmountInUsd(
     signer,
-    market.comptrollerAddress,
-    tokenPairs,
-    market.tokenPair.cToken,
-    value
+    market.tokenPair,
+    totalBorrowedAmountInUsd,
+    -value
```
Comment (ryanlabouve) [inline app/components/borrow-flow/repay.tsx:62]: Adding minus here intentional?

**Review by ryanlabouve — APPROVED** [review]
LGTM % errors. Errors look unrelated and I can handle in a followup PR

**Pull request merged**